### PR TITLE
Node check-in time series API

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Create an endpoint that creates a time series of daily counts of the number of nodes that have checked in. This has follow on work that also counts the total number of nodes for the daily time series. 

### :chains: Related Resources

https://github.com/chef/automate/issues/2995

### :athletic_shoe: How to Build and Test the Change
1. `build components/config-mgmt-service && build components/automate-gateway && start_all_services`
1. Adding some infra nodes with `chef_load_nodes 100`
1. Run this command to get the time series. `curl -s -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/cfgmgmt/stats/checkin_counts_timeseries?days_ago=3 | jq`
1. Ensure a time series of the past three days is returned. The time series should end the start of the next hour from now. 

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/config-mgmt-service/backend/elastic/runs.go [hunk 1]
```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
...
+
+	mainQuery = mainQuery.Must(rangeQuery)
+
+	bucketHist := elastic.NewDateHistogramAggregation().Field(backend.RunEndTime).
+		Interval("24h").
+		MinDocCount(0). // needed to return empty buckets
+		ExtendedBounds(
+			startTime.Format(time.RFC3339), endTime.Format(time.RFC3339)). // needed to return empty buckets
+		Format("yyyy-MM-dd'T'HH:mm:ssZ").
+		TimeZone(getTimezoneSoStartAtMidnight(startTime)). // needed start the buckets at the beginning of the day.
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/runs.go]: To force elasticsearch to start the buckets at the correct hour the time zone needs to be updated.

File: components/config-mgmt-service/backend/elastic/runs.go [hunk 2]
```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
...
+		SubAggregation(nodeID,
+			elastic.NewCardinalityAggregation().Field(backend.Id))
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/runs.go]: Counting the number of unique nodes that have ingested runs.

File: components/config-mgmt-service/backend/elastic/runs.go [hunk 3]
```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
...
+		Do(context.Background())
+	if err != nil {
+		return []backend.CheckInPeroid{}, err
+	}
+
+	numberOfHours := getNumberOfHoursBetween(startTime, endTime)
+	numberOfDays := int(math.Ceil(float64(numberOfHours) / 24.0))
+	checkInPeroids := make([]backend.CheckInPeroid, numberOfDays)
+	dateHistoRes, found := searchResult.Aggregations.DateHistogram(dateHistoTag)
+	if !found {
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/runs.go]: If there are no runs for the entire range requested `found` will be false. We do not want to return an error if this happens so we create a collection of empty buckets to return.

File: components/config-mgmt-service/backend/elastic/runs.go [hunk 4]
```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
...
+		for index := 0; index < numberOfDays; index++ {
+			checkInPeroids[index].Start = startTime.Add(time.Hour * 24 * time.Duration(index))
+			checkInPeroids[index].End = startTime.Add(time.Hour * 24 *
+				time.Duration(index)).Add(time.Hour * 24).Add(-time.Millisecond)
+		}
+		return checkInPeroids, nil
+	}
+
+	if len(dateHistoRes.Buckets) != numberOfDays {
+		return []backend.CheckInPeroid{}, errors.NewBackendError(
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/runs.go]: This should not happen but it is nice to check.

File: components/ingest-service/chef-load-data/ohai-data.json [hunk 5]
```diff
@@ -935,50 +935,6 @@
       "string_1": "Xen"
     }
   },
-  "ec2": {
```
Comment (lancewf) [inline components/ingest-service/chef-load-data/ohai-data.json]: This is removed to allow the chef load to create multiple nodes in the nodemanager. The nodemanager checks if the ec2 IDs are different for each node.

File: e2e/cypress/integration/api/cfg-mgmt/check_in_time_series.spec.ts [hunk 6]
```diff
@@ -0,0 +1,160 @@
+
+import { describeIfIAMV2p1 } from '../../../support/constants';
```
Comment (lancewf) [inline e2e/cypress/integration/api/cfg-mgmt/check_in_time_series.spec.ts]: This is a general test that goes through the load balancer.

File: api/external/cfgmgmt/cfgmgmt.proto [hunk 7]
```diff
@@ -120,6 +120,24 @@ service ConfigMgmt {
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
 
+  /*
+  GetCheckinCountsTimeSeries
+
+  Returns a daily time series of unique node check-ins for the number of day requested
```
Comment (vjeffrey) [inline api/external/cfgmgmt/cfgmgmt.proto]: day -> days

maybe add a comment about "if days ago value is empty, api will return the default 1 day ago results" ?

**Pull request merged**